frontend/src/metabase/dashboard/actions/save.js

```diff
@@ -131,13 +131,24 @@ export const updateDashboardAndCards = createThunkAction(
       });
 
       // make sure that we've fully cleared out any dirty state from editing (this is overkill, but simple)
-      dispatch(
+      await dispatch(
         fetchDashboard({
           dashId: dashboard.id,
           queryParams: null,
           options: { preserveParameters: false },
         }),
       ); // disable using query parameters when saving
+
+      // There might have been changes to dashboard card-filter wiring,
+      // which require re-fetching card data (issue #35503). We expect
+      // the fetchDashboardCardData to decide which cards to fetch.
+      dispatch(
+        fetchDashboardCardData({
+          reload: false,
+          clearCache: false,
+          // source: "SAVE",
+        }),
+      );
```

Comment:
Should we remove this `source: "SAVE"` line?

Reply:
Yeah, sorry - it's a debug leftover, will do.